src/vanilla/Templates/Rest/Common/EnumTemplate.cshtml

```diff
@@ -9,7 +9,51 @@ namespace @(Settings.Namespace).@(Settings.ModelsName)
     /// <summary>
     @WrapComment("/// ", "Defines values for " + Model.Name + ".")
     /// </summary>
-    @if (!Model.ModelAsString)
+    @if((bool)Settings.CustomSettings["ExtensibleEnums"])
+    {
+    @:[Newtonsoft.Json.JsonConverter(typeof(Microsoft.Rest.ClientRuntime.Serialization.ExtensibleEnumConverter<@Model.ClassName>))]
+    @:public sealed class @Model.ClassName : Microsoft.Rest.ExtensibleEnum<@Model.ClassName>
+    @:{
+        @foreach (var t in Model.Values)
+        {
+            if (t.Description != null)
+            {<text>
+        /// <summary>
+        @WrapComment("/// ", t.Description)
+        /// </summary>
+        </text>
+        }
+        string ctor = "new "+Model.ClassName+"(\""+t.Name+"\")";
+        @:public static readonly @Model.ClassName @t.Name = new @(Model.ClassName)("@(t.Name)");
```

Comment:
this can't really use `new` or `new` has to be smarter: If you have a predefined "red" value here, which is `new`ed, then this will not end up in the `Dictionary` of the base class (correct?). Hence, if later someone does `Create`/`Lookup` with "red", you get a *new* instance that is not referentially equal to the predefined one

Reply:
This is a static readonly field, I guess `CreateValueMap` would populate this particular field as expected.
However, I do see that we should be adding the new type to our map in `Create`, which also makes me think that it maybe should be a part of the base class itself.
Thoughts?

Reply:
The map should not be modified after it is created. The goal is to "intern" known values, not necessarily every value that is passed to `Create`.

Please refer to the usage example I gave in my comment below (it's [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/Models/AnalyzerName.cs)).